activesupport/lib/active_support/error_reporter.rb

```diff
@@ -272,5 +275,23 @@ def ensure_backtrace(error)
 
         error.backtrace.shift(count)
       end
+
+      class ErrorContextMiddlewareStack
+        def initialize
+          @stack = []
+        end
+
+        def use(middleware)
+          unless middleware.respond_to?(:call)
+            raise ArgumentError, "Error context middleware must respond to #call"
+          end
+
+          @stack << middleware
+        end
+
+        def execute(error, context)
+          @stack.each { |middleware| middleware.call(error, context) }
```

Comment:
I was thingking:

```suggestion
          @stack.inject(context) { |middleware| middleware.call(error, context) }
```

This way the contract is that the middleware return the extended context. You cna mutate it if you want but also return a different one.

It also open the door to return `false` or something like that to stop the reporting.